cabal-install/tests/IntegrationTests2.hs

```diff
@@ -118,6 +123,23 @@ main = do
           (tests config)
     )
 
+-- Tests are run silently, unless they fail. Firstly because it is annoying to
+-- see lots of stderr from your unit tests. Secondly becaus this output
+-- leaks into the result of github actions (#8419)
+silentTest :: TestTree -> TestTree
+silentTest = wrapTest silentHelper
```

Comment:
Note that `wrapTest` does quite a few things internally. Another possibility (and not incurring a dependency on `tasty-expected-failure`) would be something like 
```haskell
newtype SilentTest a = SilentTest a 

instance IsTest a => IsTest (SilentTest a) where 
  run opts (SilentTest t) yieldProgress = do 
    (out, res) <- hCapture [stderr] (run opts t yieldProgress)
    let res' = if resultSuccessful res then res else res { resultDetailsPrinter = \indent printer -> printer $ map (replicate indent ' ' ++) (lines out) }
    pure res' 
```

Reply:
I think I prefer to use `wrapTest`, since it is higher level, and other parts of the testsuite already depend on `tasty-expected-failure`. 

It's a very useful suggestion though, I didn't realise how simple it would have been to roll my own solution.